import diff viewer

nightwing opened the pull request:
https://raw.githack.com/ajaxorg/ace/78a7d8586b184e300301bcae29b0c88cf5465af4/demo/diff/index.html

nlujjawal commented:
Awesome PR diff view looks really nice. I was playing around and found some bugs

1.  alignment of fold icons is not consistent for a normal line vs a removed line.
<img width="1004" alt="Screenshot 2025-04-22 at 15 54 17" src="https://github.com/user-attachments/assets/2583b341-2277-4802-9901-eb0920587353" />
2. There is empty space in the bottom when the code is folded in the removed lines 
<img width="959" alt="Screenshot 2025-04-22 at 15 54 33" src="https://github.com/user-attachments/assets/a1c6b324-6cda-461c-a341-8f0c70253fa7" />
3. There is a very little green color bar in the removed lines fold
<img width="392" alt="Screenshot 2025-04-22 at 15 55 24" src="https://github.com/user-attachments/assets/95895463-144c-4fbc-ae3e-d7d045482f80" />

marinsokol5 commented:
Does it make sense to disable folding all together when viewing the diff? 
Cause when added lines are inside of the current content, like in the one below, folding it would hide the added line. It's kind of unclear what would be the best UX for it, so might indeed be best to not allow it at all.
<img width="619" alt="Screenshot 2025-04-23 at 10 43 48" src="https://github.com/user-attachments/assets/05f524aa-b6ad-41ff-89d0-bffb514afc71" />

We can also check how Monaco inline diff acts when folding, but I would lean on the side of disabling.

xyos commented:
Great work 🥳  I found another bug with the collapse unchanged lines option and it's related to unfolding with a similar behavior to the one @akoreman reported:

<img width="1201" alt="image" src="https://github.com/user-attachments/assets/f3450964-af49-4764-b947-01d58cce89c6" />

marinsokol5 commented:
Is dual numbering in the gutter intentional?

nightwing commented:
I have fixed most of the issues with folds except the case when fold starts outside of changed range and ends inside one, perhaps we should indeed disable folding in that case.
I do not want to disable folding altogether, since it is useful for collapsing unchanged lines

For the accept/reject feauture, i think the easiest solution will be to modify align method to include widgets for these too.

I've also added ability to select text inside the green zones, but not sure if we should limit selection to one green zone, or to allow to select as if we are in a diffview with new code editable. In principle using the approach we could make both old and new editable in inline view.
